Drivers/Src/Minions.c

```diff
@@ -0,0 +1,352 @@
+        SPI_IODIRB, 
+        rxBuf&0x40 //clear IODIRB to set as outputs except for hazard lights switch pin (bit 6, 0x40)
+    };
+    
+    // Set direction register
+    ChipSelect();
+    BSP_SPI_Write(txWriteBuf, 3);
+    ChipDeselect();
+
+    // Initialize toggle bitmap and states bitmap
+    lightToggleBitmap = 0x0000;
+    lightStatesBitmap = 0x0000;
+    
+    // Unlock mutex
+    OSMutexPost(
+        &CommMutex,
+        OS_OPT_POST_NONE,
+        &err);
+    assertOSError(OS_BLINK_LIGHTS_LOC, err);
+}
+
+/**
+ * @brief   Initializes all Lights and Switches. **Can be called only after Fault Semaphore is initialized.**
+ * @param   None
+ * @return  None
+ */ 
+void Minions_Init(void){
+    OS_ERR err;
+    OSMutexCreate(&CommMutex, "Communications Mutex", &err);
+    assertOSError(0,err);
+    BSP_SPI_Init();
+    Lights_Init();
+    Switches_Init();
+}
+
+/**
+* @brief   Read the state of a specific light from the light bitmap
+* @param   light Which Light to read
+* @return  returns State enum which indicates ON/OFF
+*/ 
+State Lights_Read(light_t light) {
+    //Return from a stored state bitmap instead of actually querying hardware. 
+    return (State) ((lightStatesBitmap>>light)&0x0001);
+}
+
+/**
+* @brief   Read the lights bitmap
+* @return  returns uint16_t with lights bitmap
+*/ 
+uint16_t Lights_Bitmap_Read() {
+    //Return from a stored state array instead of actually querying hardware.
+    return lightStatesBitmap;
+}
+
+/**
+ * @brief   Reads from static variable bitmap holding values of switches
+ * @param   sw
+ * @return  ON/OFF State
+ */ 
+State Switches_Read(switches_t sw){
+    return (State) ((switchStatesBitmap >> sw) & 0x0001);
+}
+
+/**
+ * @brief   Queries all switch-related hardware and updates our internal state with the most recent switch values 
+ * that Switches_Read depends on.
+ * 
+ */ 
+void Switches_UpdateStates(void){
+    OS_ERR err;
+    CPU_TS timestamp;
+    uint8_t query[2]={SPI_OPCODE_R,SPI_GPIOA}; //query GPIOA
+    uint8_t SwitchDataReg1 = 0, SwitchDataReg2 = 0;
+
+    //Read all switches except for ignition and hazard
+    OSMutexPend(
+        &CommMutex,
+        0,
+        OS_OPT_PEND_BLOCKING,
+        &timestamp,
+        &err
+    );
+    assertOSError(0,err);
+
+    ChipSelect();      
+    BSP_SPI_Write(query,2);
+    BSP_SPI_Read(&SwitchDataReg1,1);
+    ChipDeselect();
+    
+    //Read Hazard Switch
+    query[1] = SPI_GPIOB;
+
+    ChipSelect();
+    BSP_SPI_Write(query,2);
+    BSP_SPI_Read(&SwitchDataReg2,1);
+    ChipDeselect();
+
+    OSMutexPost(
+        &CommMutex,
+        OS_OPT_POST_NONE,
+        &err
+    );
+    assertOSError(0,err);
+
+    //Read Ignition Switch 1
+    uint8_t ign1 = BSP_GPIO_Read_Pin(PORTA, GPIO_Pin_1);
+
+    //Read Ignition Switch 2
+    uint8_t ign2 = BSP_GPIO_Read_Pin(PORTA, GPIO_Pin_0);
+    
+    //Store data in bitmap
+    switchStatesBitmap = (ign2 << 10) | (ign1 << 8) | ((SwitchDataReg2 & 0x40) << 2) | (SwitchDataReg1);
```

Comment:
```suggestion
    switchStatesBitmap = (ign2 << 10) | (ign1 << 9) | ((SwitchDataReg2 & 0x40) << 2) | (SwitchDataReg1);
```
ign1 is in the 0th bit and should be shifted to the 9th bit, so that switchStatesBitmap looks like:
10: ign2
9: ign1
8: HZD_SW
7-0: Everything else